### ISSUE TITLE
Cross contract collateral for hedged positions

### ISSUE THREAD

**Issue by Waltermelon-lang** [nexlabs22/NEX-Dex-contracts#91]
I would like you to think about the following situation and figure if this can be implemented or not:
A user want to trade EUR/GBP pair.
We have EUR/USD and GBP/USD. 
Let's say they long EUR/USD and short GBP/USD, then their position is a net long EUR/GBP.
Make sure their margin call is the combination of the both. 
Maybe we can combine all collateral in one contract and do off chain computation to check the margin for each transaction.

**Comment by Morteza3123** [comment]
We currently have this concept when user open long on bayc/usd, his bayc balance increase and his USD balance decreased.

EUR price = 1.07$
GBP price = 1.24$

Now if user wants to open 1000$ long on EUR/USD:
user EUR balance =  0 + 934 => 934
user GBP balance = 0
user USD balance = 0 - 1000 => -1000

Then user open 2000$ short on GBP/USD:
user eur balance =  934
user GBP balance = 0 - 1606 => - 1606
user USD balance = - 1000 + 2000 => +1000


 

So it's like user has long on GBP/EUR or short on EUR/GBP:
Then when user wants to close short  EUR/USD for all GBP (for example without change in the price to be simple: 

EUR price = 1.07$
GBP price = 1.24$

user EUR balance = 934 - 934 = 0
user GBP balance = -1606 + 934*(1.24/1.07) = -1606 + 1082 = -524
user USD balance = +1000

So now user has a short position on GBP/USD that can close that:

EUR price = 1.07$
GBP price = 1.24$

user EUR balance  = 0
user GBP balance =  -524 + 524
user USD balance = +1000 - 524*1.24 = 1000 - 649 = 351$

This is what I have in my mind, please share your opinion about that

**Comment by Waltermelon-lang** [comment]
Yes it works like this, my question was how to let one smartcontract know about another smartcontract position, such that the collateral can combine and the margin is effectuated on the net position of the combination. 
This would reduce the collateral requirement, reduce the margin and reduce the liquidation risk.

**Comment by Morteza3123** [comment]
For that we need to spread our exchange contract to some party contracts like collateral, position manager and so on. So position manager can manage positions according to the same collateral. But this need more time to put and implement this strategy when we have time to do that
